avaframe/com1DFAPy/SPHfunctionsCython.pyx

```diff
@@ -0,0 +1,1264 @@
+      ny: float
+          y component of the interpolated vector field at position (x, y)
+      nz: float
+          z component of the interpolated vector field at position (x, y)
+  """
+  cdef int Lx0, Ly0, Lx1, Ly1
+  cdef double Lx, Ly
+  cdef double xllc = 0.
+  cdef double yllc = 0.
+
+  # find coordinates in normalized ref (origin (0,0) and cellsize 1)
+  Lx = (x - xllc) / csz
+  Ly = (y - yllc) / csz
+
+  # find coordinates of the 4 nearest cornes on the raster
+  Lx0 = <int>Lx
+  Ly0 = <int>Ly
+  Lx1 = Lx0 + 1
+  Ly1 = Ly0 + 1
+  # prepare for bilinear interpolation
+  dx = Lx - Lx0
+  dy = Ly - Ly0
+  # using bilinear interpolation on the cell
+
+  cdef double v = V[Ly0, Lx0]*(1-dx)*(1-dy) + V[Ly0, Lx1]*dx*(1-dy) + V[Ly1, Lx0]*(1-dx)*dy + V[Ly1, Lx1]*dx*dy
+
+  return v
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+cdef (double, double, double) getVector(double x, double y, double[:, :] Nx, double[:, :] Ny, double[:, :] Nz, double csz):
+  """ Interpolate vector field from grid to single point location
+
+  Originaly created to get the normal vector at location (x,y) given the
+  normal vector field on the grid. Grid has its origin in (0,0).
+  Can be used to interpolate any vector field.
+  Interpolation using a bilinear interpolation
+
+  Parameters
+  ----------
+      x: float
+          location in the x location of desiered interpolation
+      y: float
+          location in the y location of desiered interpolation
+      Nx: 2D numpy array
+          x component of the vector field at the grid nodes
+      Ny: 2D numpy array
+          y component of the vector field at the grid nodes
+      Nz: 2D numpy array
+          z component of the vector field at the grid nodes
+      csz: float
+          cellsize of the grid
+
+  Returns
+  -------
+      nx: float
+          x component of the interpolated vector field at position (x, y)
+      ny: float
+          y component of the interpolated vector field at position (x, y)
+      nz: float
+          z component of the interpolated vector field at position (x, y)
+  """
+  cdef int Lx0, Ly0, Lx1, Ly1
+  cdef double Lx, Ly
+  cdef double xllc = 0.
+  cdef double yllc = 0.
+
+  # find coordinates in normalized ref (origin (0,0) and cellsize 1)
+  Lx = (x - xllc) / csz
+  Ly = (y - yllc) / csz
+
+  # find coordinates of the 4 nearest cornes on the raster
+  Lx0 = <int>Lx
+  Ly0 = <int>Ly
+  Lx1 = Lx0 + 1
+  Ly1 = Ly0 + 1
+  # prepare for bilinear interpolation
+  dx = Lx - Lx0
+  dy = Ly - Ly0
+  # using bilinear interpolation on the cell
+
+  cdef double nx = Nx[Ly0, Lx0]*(1-dx)*(1-dy) + Nx[Ly0, Lx1]*dx*(1-dy) + Nx[Ly1, Lx0]*(1-dx)*dy + Nx[Ly1, Lx1]*dx*dy
+  cdef double ny = Ny[Ly0, Lx0]*(1-dx)*(1-dy) + Ny[Ly0, Lx1]*dx*(1-dy) + Ny[Ly1, Lx0]*(1-dx)*dy + Ny[Ly1, Lx1]*dx*dy
+  cdef double nz = Nz[Ly0, Lx0]*(1-dx)*(1-dy) + Nz[Ly0, Lx1]*dx*(1-dy) + Nz[Ly1, Lx0]*(1-dx)*dy + Nz[Ly1, Lx1]*dx*dy
+  return nx, ny, nz
```

Comment:
multiline for readability. 
Another note: did you try to use one of numpys (or similar) interpolation routines?

Reply:
the interpolation methods in scipy did not seem faster. (there is already the numpy version of this cython function which is I think slower)